agent/src/beerocks/monitor/monitor_stats.cpp

```diff
@@ -492,6 +494,60 @@ bool monitor_stats::add_ap_metrics(ieee1905_1::CmduMessageTx &cmdu_tx, const sMa
     return true;
 }
 
+bool monitor_stats::add_ap_assoc_sta_traffic_stat(ieee1905_1::CmduMessageTx &cmdu_tx,
+                                                  monitor_sta_node *sta_node)
+{
+    auto ap_assoc_sta_traffic_stat_tlv = cmdu_tx.addClass<wfa_map::tlvAssociatedStaTrafficStats>();
+    if (!ap_assoc_sta_traffic_stat_tlv) {
+        LOG(ERROR) << "Couldn't addClass tlvAssociatedStaTrafficStats";
+        return false;
+    }
+    auto stat                                = sta_node->get_stats().hal_stats;
+    ap_assoc_sta_traffic_stat_tlv->sta_mac() = network_utils::mac_from_string(sta_node->get_mac());
+    ap_assoc_sta_traffic_stat_tlv->byte_sent()       = stat.tx_bytes;
+    ap_assoc_sta_traffic_stat_tlv->byte_recived()    = stat.rx_bytes;
+    ap_assoc_sta_traffic_stat_tlv->packets_sent()    = stat.tx_packets;
+    ap_assoc_sta_traffic_stat_tlv->packets_recived() = stat.rx_packets;
+    //TODO: add tx_packets_error in bwl::SStaStats
+    ap_assoc_sta_traffic_stat_tlv->tx_packets_error() = 0;
+    //TODO: rx_packets_error in bwl::SStaStats
+    ap_assoc_sta_traffic_stat_tlv->rx_packets_error()     = 0;
+    ap_assoc_sta_traffic_stat_tlv->retransmission_count() = stat.retrans_count;
+
+    return true;
+}
+
+bool monitor_stats::add_ap_assoc_sta_link_metric(ieee1905_1::CmduMessageTx &cmdu_tx,
```

Comment:
The equivalent of this was also done by Vitalii in #1139, so we have to find out how they interact...

Reply:
@arnout you probably wanted to say #1149.